src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -72,44 +73,75 @@ public ProductService(BaseDTO dto) {
         return components;
     }
 
+    private Set<String> getSupportedLocales() {
+        return getSupportedLocales(true);
+    }
+
+    public Set<String> getCachedSupportedLocales() {
+        return getSupportedLocales(false);
+    }
+
     /**
-     * Retrieves the list of locales of a product. It recursively applies data source fallback mechanism in case of failure.
+     * Retrieves the set of supported locales.
      *
-     * @return list of locales of the product specified in the dto object, or an empty list in case of failure to retrieve from any data source.
+     * @param refreshCache If true, it will trigger a cache populate or refresh as necessary before returning.
+     *                         If false, it will return the data from the cache as is, or an empty Set if not in cache.
+     * @return The set of supported locales.
      */
-    public Set<Locale> getSupportedLocales() {
-        return langTagtoLocaleSet(getSupportedLanguageTags());
-    }
-
-    public Set<String> getSupportedLanguageTags() {
+    private Set<String> getSupportedLocales(boolean refreshCache) {
```

Comment:
I disagree. Overloading methods is commonly done, especially to avoid duplicating code when there is very small difference between the 2 methods. I have seen this bad practice of duplicating methods and code in other places in our java client. Let's try to avoid this. This means having to maintain 2 identical code, having to make sure that if you update one, the other/s have to be updated too with the exact same change, etc.

Reply:
I have renamed this to combineSupportedLocales. The overloading should be clear. Below are the method signatures 

1. getSupportedLocales() - gets all supported locales (from all data sources)
2. getSupportedLocales(dataSource) - gets supported locales from the specified data source
3. getCachedSupportedLocales() - gets all supported locales (from all data sources) from cache
4. getCachedSupportedLocales(dataSource) - gets supported locales from the specified data source from cache
5. private combineSupportedLocales (refreshCache) - combines the list of supported locales from all dataSources. Used in # 1 and # 3.

I have also updated javadoc for all methods.